arcgis-ios-sdk-samples/Cloud and portal/Token authentication/TokenAuthenticationViewController.swift

```diff
@@ -34,6 +34,12 @@ class TokenAuthenticationViewController: UIViewController {
         return AGSMap(item: portalItem)
     }
     
+    required init?(coder aDecoder: NSCoder) {
```

Comment:
I'm in favor of not making this change, although it might cause problem when users are trying out this sample on an iPad. Ideally credential caches should be handled/cleared when a certain view controller is dismissed from view hierarchy or deinit'd. 

The unique sample viewer architecture caused a sample to persist in memory until a new sample is loaded, which makes it kind of a racing condition - old sample is de-initializing and clearing cache, and the new sample is loading from cache. This should hardly happen in a real app.

However I'm OK to add this workaround if we provide some detailed "why we add this" explanation here.

Reply:
The purpose of the samples are to demonstrate the functionality so it should do the right thing. I would suggest using the `viewDidDisappear` for OAuth and Token samples to clear the cache.

AuthenticateWithOAuthViewController - Move code from `deinit` to `viewDidDisappear` 
```
    override func viewDidDisappear(_ animated: Bool) {
        AGSAuthenticationManager.shared().oAuthConfigurations.remove(oAuthConfiguration)
        AGSAuthenticationManager.shared().credentialCache.removeAllCredentials()
    }
``

TokenAuthenticationViewController - Add code
```
    override func viewDidDisappear(_ animated: Bool) {
        AGSAuthenticationManager.shared().credentialCache.removeAllCredentials()
    }
```

Reply:
I don't think so. I can see in all places credential is either created while handling authentication challenge or set on the object before loading. That is fine.

Reply:
If the credentialCache is cleared on `viewDidDisappear`, say in the OAuth sample, when the user first logs in, then tap the upper right source code page button, the credentials will be cleared. When the source code view controller is popped, the user will see a blank screen?

Reply:
Yes, `viewDidDisappear` will be called in that case but everything will continue to work on the map. When any object used the credential to load than that credential will be available on that object's `credential` property. Only other option is to do that in init before doing anything else in both samples.

Reply:
Also, comments should be added to let user know why we are clearing the credentials.

Reply:
Thanks for the discussion! From what I've gathered, implementing `viewDidDisappear` will suffice and there's no need for `viewWillAppear`. I've made the changes accordingly. Please let me know if I understood correctly!

Reply:
> ... but everything will continue to work on the map. When any object used the credential to load than that credential will be available on that object's `credential` property. 

This is crucial info, thanks for the clarification. I was assuming by clearing the credentialCache the user's authenticated state will be nullified and cannot make further requests to the protected endpoint.